## Background

Since 2018 UMCCR has been using the [bcbio](https://github.com/bcbio/bcbio-nextgen) framework to analyze clinical sequencing data. Our [whole-genome analysis](https://github.com/umccr/workflows/blob/master/configurations/std_workflow_cancer_hg38.yaml) includes the preparation of sequecing reads, alignment to the `hg38` reference genome, ensemble-based variant calling for both germline and somatic changes, somatic structural variant calling and the generation of quality control metrics. These steps generate the primary (read-level) and secondary (variant-level) results of our workflow.

The generated results serve as input for our post-processing framework [`umccrise`](https://github.com/umccr/umccrise) which provides additional analytical steps (e.g., viral integration, somatic signatures) and generates data summaries for UMCCR's curation team -- the 'tertiary' results.

For [RNA sequence analysis](https://github.com/umccr/workflows/blob/master/configurations/std_workflow_wts_hg38.yaml) with bcbio we create both 'standard' alignments with [STAR](https://github.com/alexdobin/STAR) for visualistion and fusion detection with [Arriba](https://github.com/pachterlab/kallisto) alongside 'pseudo' alignments with [kallisto](https://github.com/pachterlab/kallisto) for transcript counts. These primary and secondary results are processed with [RNASum](https://github.com/umccr/RNAsum) to compare and contrast expression values with a reference cohort, link RNA and WGS information and to summary results.

Since the beginning of 2022 we have been testing an alternative approach to generate the primary and secondary data based on Illumina's [DRAGEN](https://www.illumina.com/products/by-type/informatics-products/dragen-bio-it-platform.html) platform deployed on Illumina's Connected Analytics (ICA V1) environment. This document describes key changes in the workflows, their environments and the impact on reference and validation samples. 

### Change Summary

The migration from bcbio to DRAGEN includes a number of key changes to our whole genome analysis:

* **Alignment:** We rely on DRAGEN read alignments instead of those generated by [bwa-mem](https://github.com/lh3/bwa). The changes between the two aligners have been [described by the GATK team](https://gatk.broadinstitute.org/hc/en-us/articles/4410953761563) at The Broad Institute who are migrating their best practice (germline) workflow to DRAGEN. DRAGEN's aligner handles ALT regions differently - it is ['ALT aware'](https://support.illumina.com/content/dam/illumina-support/help/Illumina_DRAGEN_Bio_IT_Platform_v3_7_1000000141465/Content/SW/Informatics/Dragen/GPipelineAltMap_fDG.htm) and retains links between reads on the primary assembly vs ALT regions which helps avoid false negative variant calls in these regions.
* **Read trimming:** We no longer trim Poly-X nucleotides from the end of reads, a step performed with [atropos](https://github.com/jdidion/atropos) in bcbio. This is a regression but based on our initial benchmark data the impact seems to be minimal. Poly-G sections _are_ [soft-trimmed by default](https://support-docs.illumina.com/SW/DRAGEN_v39/Content/SW/DRAGEN/ReadTrimming.htm?Highlight=trimming) which covers the most common Poly-X trimming usecase.
* **Reference genome masking:** For bcbio we limited variant calls to no-ALT regions of the reference genome; additionally, a number of difficult-to-call genomic regions were masked to prevent variant callers from taking too long. This is no longer necessary with DRAGEN, and while this results in increased false positive we reduce the risk of missing important genomic changes.
* **Germline small variant detection:** Germline SNV calls are now performed by DRAGEN's single caller instead of using a 2-out-of-3 ensemble approach (GATK Haplotype Caller, VarDict, Strelka2). This follows GATK's best practice. While the ensemble approach worked well by pairing two sensitive callers with one strict caller we continued to see problems with complex variants getting filtered out due to slightly different variant representations. The Broad Institute has demonstrated [functional equivalence](https://gatk.broadinstitute.org/hc/en-us/articles/4410456501915-Functional-equivalence-in-DRAGEN-GATK) between DRAGEN germline variant calls and GATK best practice variant calls.
* **Somatic small variant detection:** Somatic SNV calls follow the same approach, switching from a 2-out-of-3 ensemble approach (Mutect2, Vardict, Strelka2) to DRAGEN somatic variant calls. DRAGEN's variant caller is based on Strelka2 and performs better than our old ensemble approach in a comparison of [SEQC-II benchmark data](https://github.com/umccr-illumina/dragen/blob/master/notes/2020-01-09_Dragen_Update.md) while avoiding the variant representation issue outlined above.
* **Somatic structural variant detection:** Structural variant calls now use DRAGEN's SV caller instead of [Manta](https://github.com/Illumina/manta). Manta's open source development stopped in 2019 with version 1.6 and was continued as a closed-source project within DRAGEN. Based on our communication with Illumina DRAGEN's SV caller would be equivalent to Manta 1.9 if public development had continued. We benefit from [higher sensitivity](https://github.com/umccr-illumina/dragen/blob/master/notes/manta_history.md), more stringent filtering and a better integration of the somatic SNV and SV callers.
* **QC metrics:** tba


For RNA sequence analysis the changes include:

* @skanwal


### Comparison of benchmark data

All comparisons are based on bcbio 1.1.6a with umccrise 1.2.4 compared to DRAGEN 3.9 

* @alexiswl - can you add links to the current prod CWL workflows (DRAGEN Germline, DRAGEN Somatic, DRAGEN RNA) please?

#### Germline benchmark comparison

Germline comparisons are based on four replicates of NA12878 sequenced on an Illumina NovaSeq at the Melbourne Clinical Genomics Platform. Germline variant calls were compared between bcbio and DRAGEN production workflows using [vcf_eval](https://github.com/umccr/biodaily/tree/vcf_eval/vcf_eval) and are based on the [Genome in a Bottle V4.2.1 benchmark set](https://ftp-trace.ncbi.nlm.nih.gov/ReferenceSamples/giab/release/) and high confidence regions (`HG001_GRCh38_1_22_v4.2.1_benchmark.bed`).

**Sample information in the LIMS:**

| IlluminaID                    | Run | Timestamp  | SubjectID | SampleID | LibraryID | ExternalSubjectID | ExternalSampleID |
|-------------------------------|-----|------------|-----------|----------|-----------|-------------------|------------------|
| 170802_A00130_0016_AH2JGGDMXX | SBJ00027  | -        | -         | NA12878           | NA12878-1VD      |
| 170802_A00130_0016_AH2JGGDMXX | SBJ00027  | -        | -         | NA12878           | NA12878-2VD      |
| 170823_A00130_0019_AH2N2FDMXX | SBJ00027  | -        | -         | NA12878           | NA12878-3VD_S1   |
| 170829_A00130_0020_AH2MKTDMXX | SBJ00027  | -        | -         | NA12878           | NA12878-4KC_S7   |

**SNVs:**

| sample      | workflow | Truth     | TP        | FP     | FN     | Recall | Precision | f1    |
|-------------|----------|-----------|-----------|--------|--------|--------|-----------|-------|
| NA12878-1VD | bcbio    | 3,253,394 | 3,236,126 | 11,868 | 17,268 |  0.995 |     0.996 | 0.996 |
| NA12878-1VD | DRAGEN   | 3,253,394 | 3,241,892 |  7,177 | 11,502 |  0.996 |     0.998 | 0.997 |
| NA12878-2VD | bcbio    | 3,253,394 | 3,236,106 | 11,752 | 17,288 |  0.995 |     0.996 | 0.996 |
| NA12878-2VD | DRAGEN   | 3,253,394 | 3,241,517 |  6,479 | 11,877 |  0.996 |     0.998 | 0.997 |
| NA12878-3VD | bcbio    | 3,253,394 | 3,236,886 | 11,724 | 16,508 |  0.995 |     0.996 | 0.996 |
| NA12878-3VD | DRAGEN   | 3,253,394 | 3,241,927 |  6,236 | 11,467 |  0.996 |     0.998 | 0.997 |
| NA12878-4KC | bcbio    | 3,253,394 | 3,235,975 | 12,119 | 17,419 |  0.995 |     0.996 | 0.995 |
| NA12878-4KC | DRAGEN   | 3,253,394 | 3,242,203 |  7,523 | 11,191 |  0.997 |     0.998 | 0.997 |


**InDels:**

| sample      | workflow | Truth   | TP      | FP     | FN     | Recall | Precision | f1    |
|-------------|----------|---------|---------|--------|--------|--------|-----------|-------|
| NA12878-1VD | bcbio    | 468,694 | 425,709 | 65,446 | 42,985 |  0.908 |     0.867 | 0.887 |
| NA12878-1VD | DRAGEN   | 468,694 | 465,285 |  4,470 |  3,409 |  0.993 |     0.990 | 0.992 |
| NA12878-2VD | bcbio    | 468,694 | 426,549 | 65,248 | 42,145 |  0.910 |     0.867 | 0.888 |
| NA12878-2VD | DRAGEN   | 468,694 | 465,648 |  3,813 |  3,046 |  0.994 |     0.992 | 0.993 |
| NA12878-3VD | bcbio    | 468,694 | 427,611 | 67,827 | 41,083 |  0.912 |     0.863 | 0.887 |
| NA12878-3VD | DRAGEN   | 468,694 | 466,069 |  3,211 |  2,625 |  0.994 |     0.993 | 0.994 |
| NA12878-4KC | bcbio    | 468,694 | 423,757 | 66,845 | 44,937 |  0.904 |     0.864 | 0.883 |
| NA12878-4KC | DRAGEN   | 468,694 | 465,071 |  4,758 |  3,623 |  0.992 |     0.990 | 0.991 |

The DRAGEN germline SNV calls match bcbio ensemble calls and are significantly better when it comes to calling insertions or deletions, a reflection of the ensemble representation issues outlined above.


#### Somatic benchmark comparison

Somatic SNV comparisons between DRAGEN and bcbio ensemble calls are evaluated against the [SEQC-II test data](https://www.biorxiv.org/content/10.1101/625624v1) (SRA ID `SRP162370`, benchmark sets available via [FTP](ftp://ftp-trace.ncbi.nlm.nih.gov/seqc/ftp/Somatic_Mutation_WG/)). A dilution series of DNA from the SEQC-II triple-negative breast cancer (TNBC) cell line (HCC1395) and a B lymphocyte-derived normal cell line (HCC1395BL) was sequenced on an Illumina NovaSeq at the Melbourne Clinical Genomics Platform and again compared with `vcf_eval`. From the in-silico tests we noticed no differences in overall performance between 100% sample down to 50%; accordingly the dilution series sequencing was limited to the 10-40% range as we expect any problems to show up in this range.

**Sample information in the LIMS:**

| IlluminaID                    | SubjectID | SampleID       | LibraryID      | ExternalSubjectID | ExternalSampleID         |
|-------------------------------|-----|------------|-----------|----------------|----------------|-------------------|--------------------------|
| 210223_A01052_0033_AHVHWGDMXX | SBJ00480  | PTC_HCC1395_30 | L2000433_rerun | HCC1395           | Mix-cell-line-HCC1395-30 |
| 210223_A01052_0033_AHVHWGDMXX | SBJ00480  | PTC_HCC1395_10 | L2000435_rerun | HCC1395           | Mix-cell-line-HCC1395-10 |
| 210223_A01052_0033_AHVHWGDMXX | SBJ00480  | PTC_HCC1395_40 | L2000432_rerun | HCC1395           | Mix-cell-line-HCC1395-40 |
| 210223_A01052_0033_AHVHWGDMXX | SBJ00480  | PTC_HCC1395_20 | L2000434_rerun | HCC1395           | Mix-cell-line-HCC1395-20 |
| 210331_A01052_0041_BHYMHFDSXY | SBJ00480  | PTC_TsqN200511 | L2000437_rerun | HCC1395           | HCC1395BL-01             |

**SNVs:**

| sample                                 | workflow  | Truth | TP    | FP   | FN    | Recall | Precision | f1    |
|----------------------------------------|--------|-------|-------|------|-------|--------|-----------|-------|
| SBJ00480-Mix-cell-line-HCC1395-40      | bcbio  | 39447 | 31116 | 3529 |  8331 |  0.789 |     0.898 | 0.840 |
| SBJ00480-Mix-cell-line-HCC1395-40_PASS | DRAGEN | 39447 | 31179 | 1080 |  8268 |  0.790 |     0.967 | 0.870 |
| SBJ00480-Mix-cell-line-HCC1395-30      | bcbio  | 39447 | 30227 | 4228 |  9220 |  0.766 |     0.877 | 0.818 |
| SBJ00480-Mix-cell-line-HCC1395-30_PASS | DRAGEN | 39447 | 29750 |  991 |  9697 |  0.754 |     0.968 | 0.848 |
| SBJ00480-Mix-cell-line-HCC1395-20      | bcbio  | 39447 | 27345 | 3559 | 12102 |  0.693 |     0.885 | 0.777 |
| SBJ00480-Mix-cell-line-HCC1395-20_PASS | DRAGEN | 39447 | 26286 |  677 | 13161 |  0.666 |     0.975 | 0.792 |
| SBJ00480-Mix-cell-line-HCC1395-10      | bcbio  | 39447 | 15393 | 3543 | 24054 |  0.390 |     0.813 | 0.527 |
| SBJ00480-Mix-cell-line-HCC1395-10_PASS | DRAGEN | 39447 | 26286 |  677 | 13161 |  0.666 |     0.975 | 0.792 |

We see a slight loss of sensitivity for DRAGEN with a large improvement in precision, a trade-off we found to be acceptable particularly since we rescue potentially missed variants in cancer hotspots at a later stage in `umccrise`. The difference at 10% cellularity is expected as bcbio filters out variants with an allelic frequency <10% in our setting.

**InDels:**

| sample                                 | workflow  | Truth | TP   | FP   | FN   | Recall | Precision | f1    |
|----------------------------------------|--------|-------|------|------|------|--------|-----------|-------|
| SBJ00480-Mix-cell-line-HCC1395-40      | bcbio  |  1625 | 1169 | 1609 |  456 |  0.719 |     0.421 | 0.531 |
| SBJ00480-Mix-cell-line-HCC1395-40_PASS | DRAGEN |  1625 | 1193 |  591 |  432 |  0.734 |     0.669 | 0.700 |
| SBJ00480-Mix-cell-line-HCC1395-30      | bcbio  |  1625 | 1120 | 1539 |  505 |  0.689 |     0.421 | 0.523 |
| SBJ00480-Mix-cell-line-HCC1395-30_PASS | DRAGEN |  1625 | 1106 |  522 |  519 |  0.681 |     0.679 | 0.680 |
| SBJ00480-Mix-cell-line-HCC1395-20      | bcbio  |  1625 |  998 | 1410 |  627 |  0.614 |     0.414 | 0.495 |
| SBJ00480-Mix-cell-line-HCC1395-20_PASS | DRAGEN |  1625 |  933 |  360 |  692 |  0.574 |     0.722 | 0.639 |
| SBJ00480-Mix-cell-line-HCC1395-10      | bcbio  |  1625 |  503 | 1196 | 1122 |  0.310 |     0.296 | 0.303 |
| SBJ00480-Mix-cell-line-HCC1395-10_PASS | DRAGEN |  1625 |  933 |  360 |  692 |  0.574 |     0.722 | 0.639 |

As with the germline variant calls we see a marked improvement in both sensitivity and specificity for DRAGEN calls over the existing bcbio ensemble calls.

ToDo: @mhlunimelb Can you please add baseline results for SEQC-II (100%)?

### Comparison of validation samples

We routinely assess the impact of any major workflow change on the primary/secondary data as well as the generated HTML reports for a number of manually selected validation samples. Samples where chosen to reflect our usual clinical sample quality, diversity of ethnicity, gender and interesting events:

| Sample Name           | SubjectID | Sample Notes | bcbio URL | DRAGEN URL |
|-----------------------|-----------|--------------|-----------|----|
| 2016.249.17.MH.P033   | SBJ00471       | BRCA signature, BRCA2 SV      | [bcbio](https://github.com/umccr/workflows/tree/master/change_documents/2022-06-08_bcbio-to-DRAGEN/umccrise_reports/bcbio/2016.249.17.MH.P033) | [DRAGEN](https://github.com/umccr/workflows/tree/master/change_documents/2022-06-08_bcbio-to-DRAGEN/umccrise_reports/dragen/2016.249.17.MH.P033) |
| 2016.249.18.WH.P025   | SBJ00031       | Poor quality, SV40 integration, BRCA2 germline | [bcbio](https://github.com/umccr/workflows/tree/master/change_documents/2022-06-08_bcbio-to-DRAGEN/umccrise_reports/bcbio/2016.249.18.WH.P025) | [DRAGEN](https://github.com/umccr/workflows/tree/master/change_documents/2022-06-08_bcbio-to-DRAGEN/umccrise_reports/dragen/2016.249.18.WH.P025) |
| CUP-Pairs8            | n/a       | FFPE       | [bcbio](https://github.com/umccr/workflows/tree/master/change_documents/2022-06-08_bcbio-to-DRAGEN/umccrise_reports/bcbio/CUP-Pairs8) | [DRAGEN](https://github.com/umccr/workflows/tree/master/change_documents/2022-06-08_bcbio-to-DRAGEN/umccrise_reports/dragen/CUP-Pairs8) |
| SFRC01073             | SBJ00204 | BRCA signature          | [bcbio](https://github.com/umccr/workflows/tree/master/change_documents/2022-06-08_bcbio-to-DRAGEN/umccrise_reports/bcbio/SFRC01073) | [DRAGEN](https://github.com/umccr/workflows/tree/master/change_documents/2022-06-08_bcbio-to-DRAGEN/umccrise_reports/dragen/SFRC01073) |
| B_ALL_Case_10         | SBJ00040 | Blood cancer          | [bcbio](https://github.com/umccr/workflows/tree/master/change_documents/2022-06-08_bcbio-to-DRAGEN/umccrise_reports/bcbio/B_ALL_Case_10) | [DRAGEN](https://github.com/umccr/workflows/tree/master/change_documents/2022-06-08_bcbio-to-DRAGEN/umccrise_reports/dragen/B_ALL_Case_10) |
| SEQC_SEQC50           | SBJ00481 | Reference sample          | [bcbio](https://github.com/umccr/workflows/tree/master/change_documents/2022-06-08_bcbio-to-DRAGEN/umccrise_reports/bcbio/SEQC50) | [DRAGEN](https://github.com/umccr/workflows/tree/master/change_documents/2022-06-08_bcbio-to-DRAGEN/umccrise_reports/dragen/SEQC50) |
| DiploidNeverResponder | n/a       | HPV integration          | [bcbio](https://github.com/umccr/workflows/tree/master/change_documents/2022-06-08_bcbio-to-DRAGEN/umccrise_reports/bcbio/DiploidNeverResponder) | [DRAGEN](https://github.com/umccr/workflows/tree/master/change_documents/2022-06-08_bcbio-to-DRAGEN/umccrise_reports/dragen/DiploidNeverResponder) |
| SBJ00303              | SBJ00303  | ATRX frameshift          | [bcbio](https://github.com/umccr/workflows/tree/master/change_documents/2022-06-08_bcbio-to-DRAGEN/umccrise_reports/bcbio/SBJ00303)| [DRAGEN](https://github.com/umccr/workflows/tree/master/change_documents/2022-06-08_bcbio-to-DRAGEN/umccrise_reports/dragen/SBJ00303) |

Key features -- particularly those highlighted in the sample notes -- are conserved for all samples in the DRAGEN results. 

#### Data comparison

Woof placeholder

#### Report comparison

**MultiQC:**

QC Lower SNP filtering rate? Hard to tell - the old report was % filtered, this is %SNP and %InDel. Interesting to see that DRAGEN filters more in some of the reference samples (Bob, Chen) and less in others. Higher MapQ0 rate which is expected since we align to the whole genome; bcbio blocked out ALTs, repeats

QC Lower filtering rate again (despite identical filtered variants); looks like DRAGEN is generating less noise for some of these samples. WGD event recognized by Purple.

**Coverage:**
Coverage Looks identical

(tumor) HLA-A a bit better in Dragen
(normal) CDKN1C a bit better in Dragen


**CPSR:**
CPSR Identical
Identical (3 more VUS, expected)
2 extra VUS in Dragen
extra MSH2 VUS Indel in Dragen (Non-ClinVar)

**PCGR:**

PCGR Lost NDRG1 (Tier 3, 10% AF)

3 Tier4 diff between bcbio/Dragen

SEQC: MSI status flipped (from high to stable); likely due to the shift in InDel to SNV calls. There's an MSH6 mutation to support it, could be worth running past the curation team to see if they would be okay with that change; Tier 3 TTN lost which is a good thing; otherwise identical

PCGR Few extra tier 4 calls; nothing notable

NeverResponder: ost 3 inter-chromosomal SVs, going from 18 BND events to 2. I'd check the PUM2, CASC8 fusion event (2:20,251,798 to 8:127,306,881) as it has >80 reads in support in the old version to see why those have gotten filtered. Some CN shifts (lower CN difference only, can be ignored). HLA differences (!). Nice - going to trust DRAGEN on those.

Pairs8 FFPE PCGR Mostly identical. Difference in SNVs, expected (FFPE), nothing on the higher tiers.

B-All: TMB higher in Dragen (0.85 -> 0.97)
6 Tier4 extra indels in Dragen

P25: TMB 9 -> 31
VUS 10 -> 20
big diffs in somatic calls as discussed, but no diffs in Tiers 1/2
funny that both bcbio and dragen PCGR results show kataegis. See https://github.com/umccr-illumina/dragen/issues/35

**Cancer Reporter:**
SFRC01073: Reporter Looks good to me; biggest change in the SV/CNV space, everything else looks stable. Slight decrease in the number of CN segments (somatic), increase in germline; SV seems to be missing some relevant events (e.g., CREB1 BidFusG with >25 SR/PR support, NTRK1 deletions) - we need to check those. NTRK1 deletion is present in Del/Dup/Ins but not in BND - change of representation in Manta?

SEQC: Contamination detected in both reports (!); unlike PCGR the reporter has both as MSS (Stable). Another likely point in favor of the Dragen data. Very similar high HRDetect/CHORD scores, reduced number of SV calls - some with high levels of support in old data included (e.g., RNGTT deletion); CN changes again in HLA (expected), general shift of 2 CN for large chunk of chr1 (from 2 to 4). I can see some of that in the Circos plot but given the CN Noise warning I'm happy to ignore

303: Slight shift in lower tier signatures (2020); extra PTPRD BND call; Del/Dup/Ins identical for key genes with >10 reads support; sizeable # of genes with large CN differences (check LCE1D, MTX1, GBAP1, NTAN1)

Pairs 8 FFPE: Reporter Pretty much identical. More confident CHORD score (likely driven by additinal SNVs), sig 3 still there; there is no way I am going to compare SV/CNVs for an FFPE sample.

B-all: Signatures: same-ish
SNVs: few extra Tier4 indels detected in Dragen
CNVs: HEATR4/chr16/chr17 min CN diff is 2
SVs: Tier3 chr6 DEL detected in Dragen (chr6:162,198,360)


P25: Messy. TMB: increase (1.34 -> 2.24)
SVs: decrease (81 -> 24)
BNDs have.. completely vanished
CNVs: decrease (168 -> 88)
Kataegis detected in Dragen

**Additional notes:**

Homozyguous runs

https://github.com/umccr-illumina/dragen/issues/37

bcftools somatic is showing zero hom changes - likely a parsing error? They are present in bcftool stats germline


### RNA comparison

Scatterplots, selected fusions, test samples?

https://github.com/umccr/WTS_workflow_results_comparison


## Summary



## Other benefits

Setting up a bcbio analysis involves [manual and potentially error prone steps](https://github.com/umccr/google_lims/blob/master/docs/a_z_setting_up_bcbio_run.md) which have been replaced with full automation of both DRAGEN and umccrise on ICA. This results in improved audit trails, error notifications and a faster turnaround time. 

Sequencing data moves from our NovaSeq sequencers directly to ICA and is analyzed in the same environment, further reducing the turnaround time. We can now also deprecate our computational environment at the National Computational Infrastructure (NCI) which reduces cost and improves overall reliability of our workflows. 


Workflow management: https://github.com/umccr-illumina/cwl-iap


## Upcoming changes

* PoN
* FFPE filter
* Soft-trimming
